k9mail-library/src/main/java/com/fsck/k9/mail/transport/SmtpTransport.java

```diff
@@ -300,18 +308,21 @@ public void open() throws MessagingException {
             boolean authPlainSupported = false;
             boolean authCramMD5Supported = false;
             boolean authExternalSupported = false;
+            boolean authXoauth2Supported = false;
             if (extensions.containsKey("AUTH")) {
                 List<String> saslMech = Arrays.asList(extensions.get("AUTH").split(" "));
                 authLoginSupported = saslMech.contains("LOGIN");
                 authPlainSupported = saslMech.contains("PLAIN");
                 authCramMD5Supported = saslMech.contains("CRAM-MD5");
                 authExternalSupported = saslMech.contains("EXTERNAL");
+                authXoauth2Supported = saslMech.contains("XOAUTH2");
             }
             parseOptionalSizeValue(extensions);
 
-            if (mUsername != null
-                    && mUsername.length() > 0
-                    && (mPassword != null && mPassword.length() > 0 || AuthType.EXTERNAL == mAuthType)) {
+            if (TextUtils.isEmpty(mUsername)
```

Comment:
Oops. Looks like there's no tests at all on `SmtpTransport.open()`

I think I'll have a crack at making that testable (and thus this would have created a regression I think) in a separate PR.